mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -186,7 +186,11 @@ def pre_button_mark_done(self):
 
     def _create_and_assign_propagated_lot_number(self):
         for order in self:
-            if not order.is_lot_number_propagated or order.lot_producing_id:
+            if (
+                not order.is_lot_number_propagated
+                or order.lot_producing_id
+                and order.lot_producing_id.name == order.propagated_lot_producing
```

Comment:
I wonder if this test could be:

```suggestion
                or order.lot_producing_id.name == order.propagated_lot_producing
```